Automattic/wp-calypso
# New hosted site flow: remove site options step

Related to p1696297881912539-slack-C04H4NY6STW.

## Proposed Changes

Removes the site options step from the `/setup/new-hosted-site` flow. It’s a distraction, something that can be configured later by the user after they finish their purchase.

This PR also changes the site creation code so that the username isn't sent as a fallback sub-domain for the new site.

This PR works in conjunction with D129390-code which sets the newly created site's sub-domain to something the user would have gotten by using the "Generate" button in the site options step. That way new sites still have a "nerdy" sub-domain, but without having to go through the extra step.

## Testing Instructions

Apply D129390-code

Use the test store and go through the `/setup/new-hosted-site` flow after applying this code. You should not be inquired for a site title and a preferred data center.

You can test other flows to ensure their behaviour for choosing sub-domains hasn't changed. Try these for example
- `/setup/link-in-bio`
- `/setup/hundred-year-plan`
